tests/test_config.py

```diff
@@ -0,0 +1,299 @@
+# Copyright 2020-2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+import attr
+import pytest
+import sys
+
+from charmcraft.cmdbase import CommandError
+from charmcraft.config import (
+    _BasicPrime,
+    _CharmhubConfig,
+    _check_url,
+    load,
+)
+
+# Decide if we're using a Python 3.5 or older to support a jsonschema detail that uses
+# randomly ordered dictionaries
+is_py35 = (sys.version_info.major, sys.version_info.minor) < (3, 6)
+
+
+@pytest.fixture
+def create_config(tmp_path):
+    """Helper to create the config."""
+    def create_config(text):
+        test_file = tmp_path / "charmcraft.yaml"
+        test_file.write_text(text)
+        return tmp_path
+    return create_config
+
+
+# -- tests for the config loading
+
+def test_load_current_directory(create_config, monkeypatch):
+    """Init the config using charmcraft.yaml in current directory."""
+    tmp_path = create_config("""
```

Comment:
FWIW Over on Snapcraft, we heavily make use textwrap.dedent() to make sure the text blocks have correct alignment.

Reply:
Yes, used it in other places where text MUST start in column 0 (like help texts, init code templates, etc). But for YAMLs there is no need of that, so code gets cleaner.